Spigot-Server-Patches/0565-Create-HoverEvent-from-ItemStack-Entity.patch

```diff
@@ -0,0 +1,64 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: ysl3000 <yannicklamprecht@live.de>
+Date: Mon, 6 Jul 2020 22:18:04 +0200
+Subject: [PATCH] Create-HoverEvent-from-ItemStack-Entity
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index 5f5b39b3b7e56e935d1f964fbfc3385a65b7e178..52fbd28b5bb30a6c0f7d5ef15bfbbe7332d343af 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -47,6 +47,9 @@ import java.util.logging.Level;
+ import java.util.logging.Logger;
+ import javax.imageio.ImageIO;
+ //import jline.console.ConsoleReader; // Paper
++import net.md_5.bungee.api.chat.ItemTag; // Paper
++import net.md_5.bungee.api.chat.TextComponent; // Paper
++import net.md_5.bungee.api.chat.hover.content.Content; // Paper
+ import net.minecraft.server.Advancement;
+ import net.minecraft.server.ArgumentEntity;
+ import net.minecraft.server.BiomeManager;
+@@ -88,6 +91,7 @@ import net.minecraft.server.MobSpawnerPatrol;
+ import net.minecraft.server.MobSpawnerPhantom;
+ import net.minecraft.server.MobSpawnerTrader;
+ import net.minecraft.server.NBTBase;
++import net.minecraft.server.NBTTagCompound; // Paper
+ import net.minecraft.server.PlayerList;
+ import net.minecraft.server.RegistryMaterials;
+ import net.minecraft.server.RegistryReadOps;
+@@ -2375,5 +2379,35 @@ public final class CraftServer implements Server {
+     public com.destroystokyo.paper.entity.ai.MobGoals getMobGoals() {
+         return mobGoals;
+     }
++
++    @Override
++    public Content hoverContentOf(@Nonnull ItemStack itemStack) {
++        ItemTag itemTag = ItemTag.ofNbt(CraftItemStack.asNMSCopy(itemStack).getOrCreateTag().toString());
++        return new net.md_5.bungee.api.chat.hover.content.Item(
++            itemStack.getType().getKey().toString(),
++            itemStack.getAmount(),
++            itemTag);
++    }
++
++    @Override
```

Comment:
Maybe also another overload method would be nice:
hoverContentOf(Entity) which refers to the entity's already set custom name?
https://papermc.io/javadocs/paper/1.16/org/bukkit/Nameable.html#getCustomName--

Reply:
added